testing/btest/bifs/set_vector_conversions.zeek

```diff
@@ -0,0 +1,30 @@
+# @TEST-DOC: tests the vector_to_set and set_to_vector BIFs
```

Comment:
The test doesn't do what's probably the most common usage idiom given the boolean error returns:

```
local s1 = vector_to_set(v1);
if ( ! s1 )
    ... error handling ...

for ( i in s1 )
   ...
```

I don't think this would work with the dynamic typing, would it?

Reply:
```
if ( ! s1 )
    ... error handling ...
```

If you change your example to specify the type (`local s1: set[count] = vector_to_set(v1)`, assuming `v1` is a `vector of count`), then the second part is fine. The first part still fails, wanting a `bool` or `count` argument.

```
error in /Users/tim/Desktop/3059.zeek, line 4: requires an integral or boolean operand (!s)
```

Reply:
Yeah, and the challenge remains that these BIFs aren't very idiomatic for Zeek script because they work the static typing. I don't have a good solution, though, it's a shortcoming of the language.